DataManagementSystem/Agent/RequestOperations/ReplicateAndRegister.py

```diff
@@ -283,218 +287,238 @@ def ftsTransfer( self ):
       noMetaReplicas = replicas["NoMetadata"]
       noReplicas = replicas['NoReplicas']
       badReplicas = replicas['Bad']
-      noPFN = replicas['NoPFN']
 
       if validReplicas:
-        validTargets = list( set( self.operation.targetSEList ) - set( validReplicas ) )
+        validTargets = list(set(self.operation.targetSEList) - set(validReplicas))
         if not validTargets:
-          self.log.info( "file %s is already present at all targets" % opFile.LFN )
+          self.log.info("file %s is already present at all targets" % opFile.LFN)
           opFile.Status = "Done"
         else:
-          toSchedule[opFile.LFN] = [ opFile, validReplicas, validTargets ]
+          toSchedule[opFile.LFN] = [opFile, validReplicas, validTargets]
       else:
-        gMonitor.addMark( "FTSScheduleFail" )
+        gMonitor.addMark("FTSScheduleFail")
         if noMetaReplicas:
-          self.log.warn( "unable to schedule '%s', couldn't get metadata at %s" % ( opFile.LFN, ','.join( noMetaReplicas ) ) )
-          opFile.Error = "Couldn't get metadata"
+          err = "Couldn't get metadata"
+          errors[err] = errors.setdefault(err, 0) + 1
```

Comment:
when using defaultdict
```
errors[err] += 1
```

Reply:
Yes, thanks for reminding me, this is a great feature! I'll do it